### ISSUE TITLE
Aparente problema con json.loads(): No logra convertir string a diccionario

### ISSUE THREAD

**Issue by Pabloski100** [IIC2233/syllabus#694]
### Prerrequisitos
(Marcar colocando una X entre los corchetes los ítems que ya hiciste, así: "[X]")

* [x ] Leí las reglas del foro (https://github.com/IIC2233/syllabus/issues/1)
* [ x] Busqué en las *issues* si ya preguntaron mi duda y no encontré nada parecido (https://github.com/IIC2233/syllabus/issues)
* [ x] Mi duda no se trata de un tema administrativo o personal, ya que en ese caso debo contactar a mi profe, al Jefe de Bienestar o al correo del curso (https://iic2233.github.io/contacto/)
* [ x] Utilizaré un título descriptivo y llenaré correctamente esta plantilla
* [ x] De ser necesario, colocaré código simple que permita explicar mi problema o duda

### json.loads no logra convertir string a dict, aun cuando el formato es aparentemente correcto

Hola (mi duda es en relación a la tarea, pero es mas de contenido en si), estoy probando una función en la que en un momento debo convertir un str en formato json a un diccionario, el print del str junto a la clase me sale como:

`{"hola": "adios", "cosa": "a"} <class 'str'>`

Sin embargo, a la hora de hacer `json.loads(el diccinario)` ocurre el siguiente error

`json.decoder.JSONDecodeError: Extra data: line 1 column 31 (char 30)`

**¿Cuál es la causa del error?**

**Comment by dfloreaa** [comment]
Hola, por lo que tengo entendido esto es debido a que le pasas ``{"hola": "adios", "cosa": "a"} <class 'str'>`` como string a ``json.loads()``, deberías estar entregándole solamente ``{"hola": "adios", "cosa": "a"}``, avísame si eso logra arreglarlo!

**Comment by Pabloski100** [comment]
Disculpa, es que no entendí muy bien, debo pasarle ya un diccionario? Es que me estaba guiando un poco por un ejemplo del cuadernillo 2 de la semana 13, donde vi que le pasan un str y lo pasa a diccionario (veo el ejemplo justo despues de donde definen la clase Persona)

**Comment by dfloreaa** [comment]
No, disculpa la confusión, debes pasarle un string que contenga ese texto, es decir ``"{"hola": "adios", "cosa": "a"}"``

**Comment by Pabloski100** [comment]
Nop, le entrego solo la parte del string, eso solo es en un print de prueba que puse, ahora solo por si acaso probé hacer un strip() al str del diccionario por si habia problemas con espacios, pero tampoco

**Comment by aaossa** [comment]
Hola, el problema parece ser que estás tratando de deserializar con `json` algo que no es un *string* válido para JSON. En tu *issue* dijiste que le estás dando `str del diccionario` a `json.loads`, pero deberías estar usando `json.dumps(diccionario)`. Mira esto:

```python
import json

diccionario = {"hola": "adios", "cosa": "a"}

diccionario_correcto = json.loads(json.dumps(diccionario))
print(diccionario_correcto) 
# {'hola': 'adios', 'cosa': 'a'}

diccionario_incorrecto = json.loads(str(diccionario))
# json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
print(diccionario_incorrecto)
```

¿Podrías mostrar el código que estás usando? Sospecho que estás dándole la variable incorrecta o que no estás serializando apropiadamente el diccionario.

**Comment by Pabloski100** [comment]
Hola! Acabo de subir lo que hice a mi repositorio

**Comment by aaossa** [comment]
¿Podrías subirlo acá? Asumo que no es un código muy largo 😅 Si es parte de tu tarea, te recomiendo aislar lo que estás intentando hacer, así es más fácil encontrar *bugs* y errores

**Comment by Pabloski100** [comment]
Ah, ok, aquí esta, es parte de una función:

`
dict_json = bytes_mensaje_puro.decode("utf-8").strip()
dict_decodificado = json.loads(dict_json)
`

Despues de quitar unos bytes que no me sirven paso el "mensaje puro" a str, luego lo intento deserializar en un diccionario.
No se si tenga que ver, pero por si acaso también menciono que al final del "mensaje puro" hay bytes 0's, hice otra issue al respecto, pero entendio que no afectan la deserializacion.

**Comment by dfloreaa** [comment]
Mire el código que tienes en tu repositorio también para hacerme una mejor idea de tu implementación, te recomiendo revisar bien como haces el slicing del mensaje recibido, el problema va más por ahí. Fijate bien en el formato que usamos para enviar datos entre servidor y cliente, hay un por qué al hecho de pedir alguna información antes.

Saludos, espero que esto último pueda ayudarte ;)
